docs/source/upgrading_your_components.md

```diff
@@ -0,0 +1,328 @@
+check out [Updating a channel capability](./updating_capabilities.html).
+
+Note: when we use the term “upgrade” in Hyperledger Fabric, we’re referring to changing
+the version of a component (for example, going from one version of a binary to
+the next version). The term “update,” on the other hand, refers not to versions
+but to configuration changes, such as updating a channel configuration or a
+deployment script. As there is no data migration, technically speaking, in Fabric,
+we will not use the term "migration" or "migrate" here.
+
+## Overview
+
+At a high level, upgrading the binary level of your nodes is a two step process:
+
+1. Backup the ledger and MSPs.
+2. Upgrade the binaries to the latest version of Fabric.
+
+If you own both ordering nodes and peers, it is a best practice to upgrade the
+ordering nodes first. If a peer falls behind or is temporarily unable to process
+certain transactions, it can always catch up. If enough ordering nodes go down,
+by comparison, a network can effectively cease to function.
+
+This topic presumes that these steps will be performed using CLI commands. If
+you are utilizing a different deployment method (Rancher, Kubernetes, OpenShift,
+etc) consult their documentation on how to use their CLI.
+
+Note that for native and production deployments, you will also need to update
+the YAML configuration file for the nodes (for example, the `orderer.yaml` file)
+with the one from the release artifacts.
+
+To do this, backup the `orderer.yaml` or `core.yaml` file (for the peer) and
+replace it with the `orderer.yaml` or `core-yaml` file from the release artifacts.
+Then port any modified variables from the backed up `orderer.yaml` or `core.yaml`
+to the new one. Using a utility like `diff` may be helpful. Note that updating
+the YAML file from the release rather than updating your old YAML file **is the
+recommended way to update your node YAML files**, as it reduces the likelihood
+of making errors.
+
+We will talk more about these configuration files in the relevant steps.
+
+### Ledger backup and restore
+
+While we will demonstrate the process for backing up ledger data in this tutorial,
+it is not strictly required to backup the ledger data of a peer or an ordering
+node (assuming the node is part of a larger group of nodes in an ordering service).
+This is because, even in the worst case of catastrophic failure of a peer (such
+as a disk failure), the peer can be brought up with no ledger at all. You can
+then have the peer re-join the desired channels and as a result, the peer will
+automatically create a ledger for each of the channels and will start receiving
+the blocks via regular block transfer mechanism from either the ordering service
+or the other peers in the channel. As the peer processes blocks, it will also build
+up its state database.
+
+Still, some organizations may want to be more self-reliant and want to backup
```

Comment:
I don't really understand the 'self-reliant' bit.  I would say something more like:

Still, backing up ledger data enables restoring a peer without the time and computational costs associated with bootstrapping from genesis and reprocessing all transactions.

Or maybe just scratch the 'self-reliant' and just say "some organizations may want to backup their ledger data so as to..."

Reply:
Yeah that language came from the jira about backup and restore: https://jira.hyperledger.org/browse/FAB-3364?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel

Have edited it.